tests/src/org.apache.activemq/artemis-jms-client/2.28.0/src/test/java/org_apache_activemq/artemis_jms_client/ArtemisJmsClientTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_activemq.artemis_jms_client;
+
+import org.apache.activemq.artemis.core.config.Configuration;
+import org.apache.activemq.artemis.core.config.impl.ConfigurationImpl;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl;
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jms.Connection;
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ArtemisJmsClientTest {
+
+    private static final String QUEUE_NAME = "queue-test-" + new Random().nextLong();
+
+    private static final Logger logger = LoggerFactory.getLogger("ArtemisJmsClientTest");
+
+    private ActiveMQServer server;
+
+    private ActiveMQConnectionFactory connectionFactory;
+
+    @BeforeAll
+    void beforeAll() throws Exception {
+        logger.info("Starting embedded Artemis broker ...");
+        Configuration config = new ConfigurationImpl();
+        config.addAcceptorConfiguration("tcp", "tcp://127.0.0.1:61616");
```

Comment:
Why do we use exactly this port? How can we know that this port is free (if this port is taken then test will probably fail)? Shouldn't we let system to find and assign free port?

Reply:
That is default port for activemq broker. The same thing (hardcoded ports) is done in other tests too:
- https://github.com/oracle/graalvm-reachability-metadata/blob/1d44f7d8bdcba0d3a5ccbed3f9b890fe9926932e/tests/src/mysql/mysql-connector-java/8.0.29/src/test/java/mysql/MySQLTests.java#L38
- https://github.com/oracle/graalvm-reachability-metadata/blob/1d44f7d8bdcba0d3a5ccbed3f9b890fe9926932e/tests/src/org.mariadb.jdbc/mariadb-java-client/3.0.6/src/test/java/mariadb/MariaDbTests.java#L38
- https://github.com/oracle/graalvm-reachability-metadata/blob/1d44f7d8bdcba0d3a5ccbed3f9b890fe9926932e/tests/src/org.mariadb/r2dbc-mariadb/1.1.3/src/test/java/org_mariadb/r2dbc_mariadb/R2DBCMariaDBTest.java#L94
- https://github.com/oracle/graalvm-reachability-metadata/blob/1d44f7d8bdcba0d3a5ccbed3f9b890fe9926932e/tests/src/org.opengauss/opengauss-jdbc/3.1.0-og/src/test/java/org_opengauss/opengauss_jdbc/Opengauss_jdbcTest.java#L39
Could you please send me a link to an existing test which is looking for an available port?

Reply:
Sorry, this looked like a random port to me, and I didn't know this is a default port for this lib. In this case, it's fine for me. Thanks for the explanation.